tests/test_kafka.py

```diff
@@ -177,7 +82,8 @@ def test_pipeline(kerberos_flag, stop_count, jar_uri, keytab_secret, spark_servi
 
     sdk_tasks.check_running(KAFKA_SERVICE_NAME, 1, timeout_seconds=600)
 
-    consumer_config = ["--conf", "spark.cores.max=4", "--class", "KafkaConsumer"] + common_args
+    consumer_config = ["--conf", "spark.cores.max=2", "--conf", "spark.executor.cores=1",
```

Comment:
2 producers, 2 consumers - okay, looks like it is intended

Reply:
yes, it is the same here - fixed-size executors preserving specified parallelism level.